backend/src/apiserver/main.go

```diff
@@ -189,6 +195,24 @@ func loadSamples(resourceManager *resource.ResourceManager) error {
 		if configErr != nil {
 			return fmt.Errorf("Failed to decompress the file %s. Error: %v", config.Name, configErr)
 		}
+		// Patch the default bucket name read from ConfigMap
+		if common.GetBoolConfigWithDefault(HasDefaultBucketEnvVar, false) {
+			defaultBucket := common.GetStringConfig(DefaultBucketNameEnvVar)
+			projectId := common.GetStringConfig(ProjectIDEnvVar)
+			patchMap := map[string]string{
+				"<your-gcs-bucket>": defaultBucket,
+				"<your-project-id>": projectId,
+			}
+			if config.Name == "[Sample] ML - XGBoost - Training with Confusion Matrix" {
+				pipelineFile, err := server.PatchPipelineDefaultParameter(pipelineFile, patchMap)
+			}
+			if config.Name == "[Sample] Unified DSL - Taxi Tip Prediction Model Trainer" {
```

Comment:
I think that relying on sample name can be fragile. For example, I'm changing the names to `[Demo]` as part of my work on preloaded samples.

Reply:
Fixed